src/header/Header.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { StatusBar, StyleSheet, Text, View } from 'react-native';
+import isEmpty from 'lodash.isempty';
+import Icon from '../icons/Icon';
+
+export const DummyNavButton = () => (
+  <View style={styles.dummyNavBtn} />
+);
+
+export const Title = (props) => {
+  const {
+    text,
+  ...attributes,
+  } = props;
+
+  return (
+    <Text
+      numberOfLines={1}
+      {...attributes}
+    >
+      {text}
+    </Text>
+  );
+};
+
+export const NavButton = (props) => {
+  const {
+    icon,
+  ...attributes,
+  } = props;
+
+  return (
+    <Icon
+      name={icon}
+      {...attributes}
+    />
+  );
+};
+
+function generateChild(value, type) {
+  if (React.isValidElement(value)) {
+    return (
+      <View key={type}>
+        {value}
+      </View>
+    );
+  } else if (typeof value === 'object' && !isEmpty(value)) {
+    return type === 'center' ?
+      <Title {...value} key={type} /> :
+      <NavButton {...value} key={type} />;
+  }
+  return type === 'center' ? null : <DummyNavButton key={type} />;
+}
+
+function populateChildren(propChildren) {
+  const childrenArray = [];
+
+  const leftComponent = generateChild(propChildren.leftComponent, 'left');
+  const centerComponent = generateChild(propChildren.centerComponent, 'center');
+  const rightComponent = generateChild(propChildren.rightComponent, 'right');
+
+  childrenArray.push(
+    leftComponent,
+    centerComponent,
+    rightComponent,
+  );
+
+  return childrenArray;
+}
+
+const Header = (props) => {
```

Comment:
Sure, we can. In my first interation I had them split in separate files, then I changed my mind since they were very small and I could avoid unnecessary imports. Sometimes I try to balance it (like when components are really tiny), but maybe it would be better also in this case to have them separate.